experiments/standalone/Soil/full_soil_implicit.jl

```diff
@@ -0,0 +1,238 @@
+using ClimaCore
+import ClimaComms
+@static pkgversion(ClimaComms) >= v"0.6" && ClimaComms.@import_required_backends
+import SciMLBase
+import ClimaTimeSteppers as CTS
+using Plots
+using Statistics
+using DelimitedFiles
+import ClimaParams as CP
+
+using ClimaLand
+using ClimaLand.Soil
+using ClimaLand.Domains: Column
+
+rmse(v1, v2) = sqrt(mean((v1 .- v2) .^ 2))
+FT = Float64
+ν = FT(0.395)
+ν_ss_quartz = FT(0.92)
+ν_ss_om = FT(0.0)
+ν_ss_gravel = FT(0.0)
+Ksat = FT(4.42 / 3600 / 100) # m/s
+S_s = FT(1e-3) #inverse meters
+vg_n = FT(1.89)
+vg_α = FT(7.5) # inverse meters
+hydrology_cm = vanGenuchten{FT}(; α = vg_α, n = vg_n)
+θ_r = FT(0.0)
+params = Soil.EnergyHydrologyParameters(
+    FT;
+    ν,
+    ν_ss_om,
+    ν_ss_quartz,
+    ν_ss_gravel,
+    hydrology_cm,
+    K_sat = Ksat,
+    S_s,
+    θ_r,
+);
+
+zmax = FT(0)
+zmin = FT(-1.0)
+nelems = 50
+soil_domain = Column(; zlim = (zmin, zmax), nelements = nelems);
+surface_water_flux = WaterFluxBC((p, t) -> 0.0)
+bottom_water_flux = WaterFluxBC((p, t) -> 0.0);
+
+surface_heat_flux = HeatFluxBC((p, t) -> 0.0)
+bottom_heat_flux = HeatFluxBC((p, t) -> 0.0);
+
+boundary_fluxes = (;
+    top = WaterHeatBC(; water = surface_water_flux, heat = surface_heat_flux),
+    bottom = WaterHeatBC(; water = bottom_water_flux, heat = bottom_heat_flux),
+);
+
+sources = (PhaseChange{FT}(),);
+
+soil = Soil.EnergyHydrology{FT}(;
+    parameters = params,
+    domain = soil_domain,
+    boundary_conditions = boundary_fluxes,
+    sources = sources,
+);
+
+exp_tendency! = make_exp_tendency(soil);
+imp_tendency! = make_imp_tendency(soil);
+tendency_jacobian! = ClimaLand.make_tendency_jacobian(soil);
+
+
+function init_soil!(Y, z, Trange, params)
+    ν = params.ν
+    θ_r = params.θ_r
+    FT = eltype(Y.soil.ϑ_l)
+    zmax = FT(0)
+    zmin = FT(-1)
+
+    theta_max = FT(ν * 0.5)
+    theta_min = FT(ν * 0.4)
+    (T_max, T_min) = Trange
+
+    c = FT(20.0)
+    @. Y.soil.ϑ_l =
+        theta_min +
+        (theta_max - theta_min) * exp(-(z - zmax) / (zmin - zmax) * c)
+    Y.soil.θ_i .= FT(0.0)
+
+    T = @.(T_min + (T_max - T_min) * exp(-(z - zmax) / (zmin - zmax) * c))
+
+    θ_l = Soil.volumetric_liquid_fraction.(Y.soil.ϑ_l, ν, θ_r)
+    ρc_s =
+        Soil.volumetric_heat_capacity.(
+            θ_l,
+            Y.soil.θ_i,
+            params.ρc_ds,
+            params.earth_param_set,
+        )
+    Y.soil.ρe_int .=
+        Soil.volumetric_internal_energy.(
+            Y.soil.θ_i,
+            ρc_s,
+            T,
+            params.earth_param_set,
+        )
+end
+set_initial_cache! = make_set_initial_cache(soil);
+
+stepper = CTS.ARS111()
+err = (FT == Float64) ? 1e-8 : 1e-4
+convergence_cond = CTS.MaximumError(err)
+conv_checker = CTS.ConvergenceChecker(norm_condition = convergence_cond)
+ode_algo = CTS.IMEXAlgorithm(
+    stepper,
+    CTS.NewtonsMethod(
+        max_iters = 50,
+        update_j = CTS.UpdateEvery(CTS.NewNewtonIteration),
+        convergence_checker = conv_checker,
```

Comment:
do we want to just use small `max_iters` and remove `convergence_checker` here too?

Reply:
I think here I want to have it use the convergence checker, since we are testing conservation, and then this serves as sort of a best case scenario. In reality we'll use a smaller max_iters and not use the convergence checker at all, like you're alluding to.